.evergreen/functions.yml

```
@@ -448,6 +457,54 @@ functions:
           echo "Uploading release assets to S3 and GitHub if needed..."
           npm run --workspace mongodb-compass upload
 
+  publish-dev:
+    - command: shell.exec
+      params:
+        working_dir: src
+        shell: bash
+        env:
+          <<: *compass-env
+        script: |
+          set -e
+          set -x
+          # Load environment variables
+          eval $(.evergreen/print-compass-env.sh)
+          
+          TMP_FILE=$(mktemp)
+          BUCKET_URL="https://mciuploads.s3.amazonaws.com/${project}/compass/dev"
+          FILE_NAME="release.json"
+          VERSION_ID=$(awk '{gsub("_", ""); print}' <<< ${created_at})
+
+          if [[ "${project}" != "10gen-compass-main" ]]; then
+            echo "Trying to publish main compass (dev build) from ${project} project";
+            exit 1;
+          fi
+
+          code=$(curl -sSf "$BUCKET_URL/$FILE_NAME" -o "$TMP_FILE" -w "%{http_code}" || true)
+          if [[ $code -eq 200 ]]; then
+            CURRENT_VERSION_ID=$(jq ".id" $TMP_FILE)
+            echo "Current published version: ${CURRENT_VERSION_ID}"
+
+            # Check if the current version is greater than the version we are trying to publish.
+            if [[ $VERSION_ID -le $CURRENT_VERSION_ID ]]; then
+              echo "Published version ${CURRENT_VERSION_ID} is greater than version ${VERSION_ID} trying to publish. Exiting..."
+              exit 1
+            fi
+          fi
+
+          VERSION="v0.0.0-dev.$VERSION_ID"
+          PUBLISHED_AT=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
+          echo "{\"id\": $VERSION_ID, \"version\": \"v0.0.0-dev.$VERSION_ID\", \"published_at\": \"$PUBLISHED_AT\", \"draft\": true}" > $FILE_NAME
```

do we need quotes to surround the version?

```suggestion
          echo "{\"id\": \"$VERSION_ID\", \"version\": \"v0.0.0-dev.$VERSION_ID\", \"published_at\": \"$PUBLISHED_AT\", \"draft\": true}" > $FILE_NAME
```

do we need a json here? how we would use these info, if they are not useful (especially the draft: true), we could rather have https://mciuploads.s3.amazonaws.com/${project}/compass/dev/LATEST and the content is just the version

---

i dont have a strong preference here, but i wanted to keep the published datetime in the json. we can totally drop everything and just have the versionId in the file - `0.0.0-dev.YYDDMMHHMMSS` format and then use that in update api.